repo2docker/buildpacks/julia/julia_require.py

```diff
@@ -0,0 +1,170 @@
+"""Generates a Dockerfile based on an input matrix with REQUIRE for legacy Julia"""
+import os
+from ..python import PythonBuildPack
+
+
+class JuliaRequireBuildPack(PythonBuildPack):
+    """
+    Julia build pack which uses conda and REQUIRE.
+    """
+
+    minor_julias = {
+        '0.6': '0.6.4',
+        '0.7': '0.7.0',
+        '1.0': '1.0.3',
+        '1.1': '1.1.0',
+    }
+    major_julias = {
+        '1': '1.1.0',
+    }
```

Comment:
Could we extend this to include all the previous julia versions, as I'd done here?: https://github.com/jupyter/repo2docker/compare/master...NHDaly:julia_projecttoml#diff-fcf7bb456679b31cc83386be059cb3abR167

It's nice to be able to share a benchmark on an older version when investigating a performance regression, for example.
But I can also see the argument for wanting to avoid the maintenance burden of supporting these old julia versions for forever...

Reply:
I would suggest we could do that in a new PR. Right now the REQUIRE file was just moved around in this PR, with no functionality added or removed. I think adding new functionality (like supporting more julia versions) can be a new PR once this one here is merged.

I think supporting all old julia versions might not be easy, because IJulia needs to work for all of these and I'm kind of doubtful that one can just write one generic setup code for that.

I actually also think that adding support for more old julia versions is not a good idea in general. That stuff is really legacy, was never supported here, and so I don't really see any serious use for it, with a whole lot of maintenance work.

But at the end of the day, I guess I'd be fine with either solution, as long as it comes as a new PR :)

Reply:
👍 to a new issue/PR when it comes to changing the behaviour of what was already there. Just from this short discussion here it sounds like something we should ponder for a bit.

My test for punting: "is this PR better than what we have now and would one of the outcomes of the discussion lead to breakage in this PR?" in this case I think the answer yes and no -> punt. (What is the logic operator for yes and no -> yes :) )